### PR TITLE
[CSPM] Run commands as nobody by default

### PR DESCRIPTION
**Pull request by lebauce**
### Motivation

Currently, the `command` uses the current user to run command.
This PR allows using a different user to run commands and defaults to `nobody`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by sgnn7 — APPROVED** [review]
LGTM for agent-core owned file (though I see a ton of build failures)

File: pkg/compliance/checks/command_utils_unix.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
```
Comment (paulcacheux) [inline pkg/compliance/checks/command_utils_unix.go:6]: I think this is a >1.17 build constraint

File: pkg/compliance/checks/command_utils_windows.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+//go:build windows
```
Comment (paulcacheux) [inline pkg/compliance/checks/command_utils_windows.go:8]: I think this is a >1.17 build constraint

File: pkg/compliance/checks/command_utils_unix.go (hunk 1 above)
Comment (djmitche) [reply pkg/compliance/checks/command_utils_unix.go:6]: I think editors, or some go tool, are adding this automatically?  I see it added in lots of PRs.

Comment (paulcacheux) [reply pkg/compliance/checks/command_utils_unix.go:6]: I believe it's `gofmt` from go 1.17. It's harmless and we'll need to fix all constraints when migrating to 1.17 so this doesn't seem blocking to me

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**